lib/core/networking/server.dart

```diff
@@ -0,0 +1,156 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'websockets/session.dart';
+import 'websockets/middleware.dart';
+
+class WebServer {
+  /// Awaitable to ensure ready for use
+  late Future<void> ready;
+
+  /// List of middleware that should be installed in each WebSocket session.
+  ///
+  /// Each middleware recieves the calling [Session] instance, message,
+  /// [MiddlewareAction] and a proxy to the next [MiddlewareFunc].
+  List<MiddlewareFunc> wsMiddleware;
+
+  /// Handler for received WebSocket messages
+  Future<dynamic> Function(ServerSession, dynamic)? wsOnMessage;
+
+  /// Handler for encountered WebSocket errors
+  void Function(ServerSession, Object)? wsOnError;
+
+  /// Handler for stream WebSocket closure
+  void Function(ServerSession)? wsOnDone;
+
+  /// Handler for received HTTP requests.
+  ///
+  /// NOTE: Please ensure this calls `close()` on the response
+  void Function(HttpRequest)? httpOnRequest;
+
+  /// If WebSocket upgrade requests should be redirected
+  /// to the HTTP handler if no [wsOnMessage] was specified.
+  bool redirectWebsocket;
+
+  /// Handler for stream WebSocket closure that wraps [wsOnDone]
+  /// but also removes the session from the server.
+  late void Function(ServerSession) wsOnDoneTeardown;
+
+  /// HttpServer instance used to establish initial connection
+  HttpServer? _server;
+
+  /// List of all connected WebSockets
+  final List<ServerSession> _wsSessions = <ServerSession>[];
+
+  /// Create a WebServer that can handle HTTP and WebSocket connections
+  ///
+  /// For a secure connection specify a [securityContext] the server should use.
+  WebServer(
+    InternetAddress bindAddress,
+    int bindPort, {
```

Comment:
Named? We need an int, we ask for an int? In this context `bindPort` should be really clear, especially if it comes after `bindAddress`.

Reply:
```dart 
WebServer(InternetAdress.loopbackV4, 0)
```

vs. 

```dart 
WebServer(InternetAdress.loopbackV4, port: 0)
```

`0` itself has no special meaning. You can, oc, guess that this has to be the port, but why do you want to guess, when it can be precise as well?

Reply:
Resolved